fpm/src/fpm_compiler.f90

```diff
@@ -46,206 +25,302 @@ subroutine add_compile_flag_defaults(build_name,compiler,model)
+use fpm_filesystem, only: join_path, basename
+implicit none
+public :: get_module_flags
+public :: get_default_compile_flags
+public :: get_debug_compile_flags
+public :: get_release_compile_flags
+
+enum, bind(C)
+    enumerator :: &
+        id_unknown, &
+        id_gcc, &
+        id_f95, &
+        id_caf, &
+        id_intel_classic, &
+        id_intel_llvm, &
+        id_pgi, &
+        id_nvhpc, &
+        id_nag, &
+        id_flang, &
+        id_ibmxl, &
+        id_cray, &
+        id_lahey, &
+        id_lfortran
+end enum
+integer, parameter :: compiler_enum = kind(id_unknown)
+
+contains
+
+subroutine get_default_compile_flags(compiler, release, flags)
+    character(len=*), intent(in) :: compiler
+    logical, intent(in) :: release
+    character(len=:), allocatable, intent(out) :: flags
+    integer :: id
+
+    id = get_compiler_id(compiler)
+    if (release) then
+        call get_release_compile_flags(id, flags)
+    else
+        call get_debug_compile_flags(id, flags)
+    end if
 
+end subroutine get_default_compile_flags
+
+subroutine get_release_compile_flags(id, flags)
+    integer(compiler_enum), intent(in) :: id
+    character(len=:), allocatable, intent(out) :: flags
+
+    select case(id)
+    case default
+        flags = ""
+
+    case(id_caf)
+        flags='&
+            & -O3&
+            & -Wimplicit-interface&
+            & -fPIC&
+            & -fmax-errors=1&
+            & -funroll-loops&
+            &'
+    case(id_gcc)
+        flags='&
+            & -O3&
+            & -Wimplicit-interface&
+            & -fPIC&
+            & -fmax-errors=1&
+            & -funroll-loops&
+            & -fcoarray=single&
+            &'
+    case(id_f95)
+        flags='&
+            & -O3&
+            & -Wimplicit-interface&
+            & -fPIC&
+            & -fmax-errors=1&
+            & -ffast-math&
+            & -funroll-loops&
+            &'
+    case(id_nvhpc)
+        flags = '&
+            & -Mbackslash&
+            &'
+    case(id_intel_classic)
+        flags = '&
+            & -fp-model precise&
+            & -pc 64&
+            & -align all&
+            & -error-limit 1&
+            & -reentrancy threaded&
+            & -nogen-interfaces&
+            & -assume byterecl&
+            &'
+    case(id_nag)
+        flags = ' &
+            & -O4&
+            & -coarray=single&
+            & -PIC&
+            &'
+    end select
+end subroutine get_release_compile_flags
+
+subroutine get_debug_compile_flags(id, flags)
+    integer(compiler_enum), intent(in) :: id
+    character(len=:), allocatable, intent(out) :: flags
+
+    select case(id)
     case default
-       fflags = ' '
-       mandatory=' -module '//modpath//' -I '//modpath 
-       write(*,'(*(a))')'<WARNING> unknown compiler (',compiler,') and build name (',build_name,') combination.'
-       write(*,'(a,*(T31,6(a:,", "),/))')'          known compilers are ',(trim(names(i)),i=1,size(names)-1)
```

Comment:
Is there a reason we no longer want to warn about unknown compiler? I would suggest adding it at the end of `get_compiler_id`.